flink-core/src/main/java/org/apache/flink/api/common/serialization/SerializerConfig.java

```diff
@@ -352,6 +361,20 @@ public void configure(ReadableConfig configuration, ClassLoader classLoader) {
                 .getOptional(PipelineOptions.KRYO_REGISTERED_CLASSES)
                 .map(c -> loadClasses(c, classLoader, "Could not load kryo type to be registered."))
                 .ifPresent(c -> this.registeredKryoTypes = c);
+
+        try {
+            configuration
+                    .getOptional(PipelineOptions.SERIALIZATION_CONFIG)
+                    .ifPresent(c -> parseSerializationConfigWithExceptionHandling(classLoader, c));
+        } catch (Exception e) {
+            if (!GlobalConfiguration.isStandardYaml()) {
+                throw new UnsupportedOperationException(
+                        String.format(
+                                "%s is only supported with the standard YAML config parser, please use \"config.yaml\" as the config file.",
+                                PipelineOptions.SERIALIZATION_CONFIG.key()));
+            }
```

Comment:
Shall we add the Exception e to UnsupportedOperationException object as root cause?

Reply:
Seems unnecessary, we are certain that it won't work when using the legacy parser. So, this is already a concise root cause